faultplugin/faultplugin.c

```diff
@@ -1327,6 +1335,44 @@ int readout_control_qemu(void)
 	return 0;
 }
 
+static struct sigaction old_abort_action;
+
+static void handle_abort(int no, siginfo_t * info, void *context)
+{
+	qemu_plugin_outs("[DEBUG]: Abort handler called\n");
+	g_autoptr(GString) end_reason = g_string_new("QEMU aborted");
+	plugin_end_information_dump(end_reason);
+	if (old_abort_action.sa_handler != SIG_DFL && old_abort_action.sa_handler != SIG_IGN) 
```

Comment:
I did it this way as I was concerned that calling raise() again would result in the old abort signal handler receiving different parameters for info and context than it would have if this plugin abort handler had not been registered. It would probably be ok without this logic as there doesn't appear to be any custom SIGABRT handlers in the QEMU code besides test code. However, I was considering the possibility of other plugins being added and running an parallel with this one.

Regarding the default action not running on raise or kill, I don't think the expectation is that the default action would always run via raise() or kill(). Only the documentation for abort() explicitly states the default action will be run after any custom signal handlers are executed. In other words, raise(SIGABRT) and kill(x, SIGABRT) do not guarantee the process will terminate.

Reply:
You are right, only abort() guarantees the default action is run. I agree with you, it makes sense to call the old abort signal handler as in the pull request to ensure it receives the same input values.

Thanks for explaining your reasoning behind this!